browse.html

```diff
@@ -51,56 +70,115 @@
+    objectResults.push(object);
+    drawCard(object);
+
+    if (i === startIndex) {
+      loader.style.display = 'none';
+      grid.style.display = 'flex';
+    }
+  }
+
+  pageNumEl.innerHTML = `Page ${currentPage + 1}/${totalPages}`;
+  pagination.style.display = 'flex';
+}
+
+async function getObjectByIndex(index) {
+  const metadataHash = await contract.methods.getMetadata(index, 'hash').call();
+  const metadata = await fetch(`${apiHost}/${metadataHash}`).then(res => res.json());
+  const {dataHash, screenshotHash, modelHash} = metadata;
+  return {
+    index: index,
+    name: metadata.objectName,
+    img: `${apiHost}/${screenshotHash}`,
+    metadataHash: metadataHash,
+    dataHash: dataHash,
+    modelHash: modelHash
+  }
+}
+
 // contract.init();
 
 const gridEl = document.getElementById('grid');
 (async () => {
   let nonce = await contract.methods.getNonce().call();
-  nonce = parseInt(nonce, 10);
-
-  for (let i = 1; i <= nonce; i++) {
-    const metadataHash = await contract.methods.getMetadata(i, 'hash').call();
-    const metadata = await fetch(`${apiHost}/${metadataHash}`)
-      .then(res => res.json());
-    const {dataHash, screenshotHash, modelHash} = metadata;
-    /* const [
-      data,
-      screenshotImg,
-    ] = await Promise.all([
-      fetch(`${apiHost}/${dataHash}`)
-        .then(res => res.arrayBuffer()),
-      new Promise((accept, reject) => {
-        const img = new Image();
-        img.src = `${apiHost}/${screenshotHash}`;
-        img.onload = () => {
-          accept(img);
-        };
-        img.onerror = reject;
-      }),
-    ]); */
-    const div = document.createElement('div');
-    div.classList.add('object');
-    div.classList.add('card');
-    div.innerHTML = `\
-      <a href="/run.html?i=${i}">
-        <img src="${apiHost}/${screenshotHash}" width=256 height=256>
-      </a>
-      <div class=text>
-        <div class="name cardTitle">${metadata.objectName}</div>
-        <input type=text value="xrpk install ${i}" readonly class="cardCode">
-        <a href="/run.html?i=${i}" class="cardAction"><span>Run</span><i class="fa fa-chevron-right"></i></a>
-        <a href="https://cryptopolys.com/create.html?o=${encodeURIComponent(metadataHash)}" class="cardAction"><span>Edit</span><i class="fa fa-chevron-right"></i></a>
-        <a href="https://ipfs.exokit.org/ipfs/${dataHash}.wbn" class="cardAction"><span>Download package</span><i class="fa fa-chevron-right"></i></a>
-        <a href="https://ipfs.exokit.org/ipfs/${modelHash}.glb" class="cardAction"><span>Download model</span><i class="fa fa-chevron-right"></i></a>
-        <a href="https://${network}.opensea.io/assets/${address}/${i}" class="cardAction"><span>Opensea</span><i class="fa fa-chevron-right"></i></a>
-      </div>
-    `;
-    const input = div.querySelector('input');
-    input.addEventListener('click', e => {
-      input.select();
-    });
-    grid.appendChild(div);
+  nonce, totalObjects = parseInt(nonce, 10);
```

Comment:
I don't understand this line; what is the `nonce,` for?

Reply:
That was leftover from the previous implementation, fixed in 1d7b945